ddtrace/internal/telemetry/writer.py

```diff
@@ -381,7 +394,13 @@ def _app_started_event(self):
             "dependencies": get_dependencies(),
             "integrations": self._flush_integrations_queue(),
             "configurations": [],
+            "error": {
```

Comment:
what happens if we get the error after the app-started event?

e.g. we likely will submit the app-started payload before we try to send traces to the agent for the first time

Reply:
Will the telemetry writer not write this then? https://github.dev/DataDog/dd-trace-py/pull/5500/files#diff-0cb9315daf5ed5208de657467cc10a6b7dda616a161c9b5b1f8cff83c95894deR129

Reply:
We only ever call `_app_started_event()` from `_start_service`. And we start the telemetry writer when we start the agent writer.

It seems like we delay start the telemetry writer? We only enable it when the trace writer is enabled?

https://github.com/DataDog/dd-trace-py/blob/a21545212/ddtrace/internal/writer.py#L695-L699

So my concerns might not be an issue.

I do hate the explicit startup relationship here, but... should be ok for now how it is written.

Reply:
Ah Kyle explained. Understood.